client/node-common/src/command.rs

```diff
@@ -42,19 +42,5 @@ pub fn generate_genesis_block<Block: BlockT>(
         genesis_state_version,
     );
 
-    let extrinsics_root = <<<Block as BlockT>::Header as HeaderT>::Hashing as HashT>::trie_root(
-        Vec::new(),
-        genesis_state_version,
-    );
-
-    Ok(Block::new(
-        <<Block as BlockT>::Header as HeaderT>::new(
-            Zero::zero(),
-            extrinsics_root,
-            state_root,
-            Default::default(),
-            Default::default(),
-        ),
-        Default::default(),
-    ))
+    Ok(construct_genesis_block(state_root, genesis_state_version))
```

Comment:
is this doing effectively the same thing?

Reply:
Yes, I tried to refactor it further to avoid trying both v0 and v1 state, but I gave up because the way we do it now is probably faster. But I left this because it seemed duplicated code from polkadot-sdk.